app/controllers/merchants_controller.rb

```diff
@@ -0,0 +1,56 @@
+class MerchantsController < ApplicationController
+  before_action :require_user, only: [:edit, :show]
+  before_action :redirect_if_logged_in, only: [:new, :create]
+
+  def index
+  end
+
+  def new
+    @merchant = Merchant.new
+  end
+
+  def create
+  @merchant = Merchant.create(merchant_params)
+    if @merchant.save
+# makes it so they don't have to login after they sign up. Takes the session data so it can run the redirect.
+      session[:merchant_id] = @merchant.id
+      redirect_to merchant_path(@merchant)
+    else
+      flash[:error] = "Try again. Or you may already be a user. Try logging in."
+      render :new
+    end
+  end
+
+  def show
+    id = @current_user.id
+    @merchant = Merchant.find(id)
```

Comment:
We did want to make it so that the merchant would only be allowed to see their own show page.  Can you clarify if you see a bug or unnecessary code? Or are you just saying that you think you can tell what we're trying to do?